pyro/infer/mcmc/adaptation.py

```diff
@@ -8,9 +8,11 @@
 
 import pyro
 from pyro.ops.dual_averaging import DualAveraging
-from pyro.ops.welford import WelfordCovariance
+from pyro.ops.welford import WelfordCovariance, WelfordArrowheadCovariance
 
 adapt_window = namedtuple("adapt_window", ["start", "end"])
+ArrowheadMatrix = namedtuple("ArrowheadMatrix", ["top", "bottom_diag"])
+ArrowheadUpperSqrtMatrix = namedtuple("ArrowheadUpperSqrtMatrix", ["top", "bottom_diag"])
```

Comment:
Consider moving these namedtuples and the math helpers to a file `pyro.ops.arrowhead`. I think these could be marked `EXPERIMENTAL`. Then you could `import pyro.ops.arrowhead as arrowhead`.

Reply:
i agree that it would be nice to move some of this to `ops` 👍

Reply:
I have moved those stuffs to ops and renamed the lengthy names into `triu_inverse`, `triu_matvecmul`, `triu_gram` (because those maths are true for general positive definite triu arrowhead matrices).